seed/analysis_pipelines/pipeline.py

```diff
@@ -132,13 +249,12 @@ def start_analysis(self):
 
         return self._start_analysis()
 
-    def fail(self, message, progress_data_key=None, logger=None):
-        """Fails the analysis. Creates an AnalysisMessage and optionally logs it
-        if a logger is provided.
+    def fail(self, message, logger, progress_data_key=None):
```

Comment:
Was curious about this: "Where's this used? And could that react badly with a non-existent Analysis?" I can answer the first part myself with some more research, but figured I'd pose it here in case it was an easy answer for you.

Reply:
> Where's this used

This is used within pipeline tasks to "fail" the analysis. bsyncr pipeline uses it if it's not configured to run bsyncr, or for some of the steps it will fail if it is unable to do work for any of the property views (e.g. none have meter data/lat/lng)

> And could that react badly with a non-existent Analysis?

This could cause raise a Analysis.DoesNotExist exception if the analysis doesn't exist.
If called within a pipeline task, and it's not caught (which is how we should call it), then the exception will be logged and 'swallowed' by the analysis_pipeline_task wrapper.
If it's called outside of a pipeline task (like we do in `BsyncrPipeline._prepare_analysis`), then the caller will need to catch it.

I think we should chat about how we should handle this